test/python/circuit/test_circuit_operations.py

```diff
@@ -948,6 +949,23 @@ def test_compare_a_circuit_with_none(self):
 
         self.assertFalse(qc1 == qc2)
 
+    def test_add_existing_bits(self):
+        """Test add registers whose bits have already been added."""
+        qc = QuantumCircuit()
+        for bit_type, reg_type in (
+            [Qubit, QuantumRegister],
+            [Clbit, ClassicalRegister],
+            [AncillaQubit, AncillaRegister],
+        ):
+            bits = [bit_type() for _ in range(10)]
+            reg = reg_type(bits=bits)
+            qc.add_bits(bits)
+            qc.add_register(reg)
+
+        self.assertEqual(qc.num_qubits, 20)
+        self.assertEqual(qc.num_clbits, 10)
+        self.assertEqual(qc.num_ancillas, 10)
+
```

Comment:
Please could we add a test that calling `add_register` with two different `AncillaRegister`s containing the same bits, without calling `add_bits` at all, also adds the correct number of bits?  With this PR it will succeed, because `AncillaRegister` derives from `QuantumRegister`, but it'd be good to ensure that it's always the case - we can't rely on `add_bits` having been called by the user first.

Reply:
I see. It has been added with the name `test_overlapped_add_register_and_add_register`